tests/test_jsonl.py

```diff
@@ -0,0 +1,199 @@
+        assert_document_counts(documents, 1, 2, 1, line_semantics)
+    # Two documents
+    for line_semantics in two_documents_one_paragraph_semantics:
+        documents = list(read_documents(test_document, line_sematics=line_semantics, lang=lang))
+        assert_document_counts(documents, 2, 1, 1, line_semantics)
+    assert len(LineSemantics) == len(
+        one_document_two_paragraph_semantics
+        + one_document_two_sentences_semantics
+        + two_documents_one_paragraph_semantics
+    )
+
+
+def test_read_documents_two_lines_with_empty_lines():
+    test_document = ["test sentence.\n", "\n", "\n", "Test 3.\n"]
+    lang = "en"
+    one_document_two_sentences_semantics = [
+        LineSemantics.sent_ignore,
+    ]
+    one_document_two_paragraph_semantics = [
+        LineSemantics.sent_para,
+        LineSemantics.para_ignore,
+    ]
+    two_documents_one_paragraph_semantics = [
+        LineSemantics.sent_doc,
+        LineSemantics.sent_para_doc,
+        LineSemantics.para_doc,
+        LineSemantics.para_ignore_doc,
+        LineSemantics.doc_ignore,
+    ]
+    for line_semantics in one_document_two_sentences_semantics:
+        documents = list(read_documents(test_document, line_sematics=line_semantics, lang=lang))
+        assert_document_counts(documents, 1, 1, 2, line_semantics)
+    for line_semantics in one_document_two_paragraph_semantics:
+        documents = list(read_documents(test_document, line_sematics=line_semantics, lang=lang))
+        assert_document_counts(documents, 1, 2, 1, line_semantics)
+    # Two documents
+    for line_semantics in two_documents_one_paragraph_semantics:
+        documents = list(read_documents(test_document, line_sematics=line_semantics, lang=lang))
+        assert_document_counts(documents, 2, 1, 1, line_semantics)
+    assert len(LineSemantics) == len(
+        one_document_two_paragraph_semantics
+        + one_document_two_sentences_semantics
+        + two_documents_one_paragraph_semantics
+    )
+
+
+def test_read_documents_line_segmentation():
+    test_document = ["test sentence. Second sentence.\n"]
+    lang = "en"
+    one_sentence = [
+        LineSemantics.sent_ignore,
+        LineSemantics.sent_para,
+        LineSemantics.sent_doc,
+        LineSemantics.sent_para_doc,
+    ]
+    two_sentences = [
+        LineSemantics.para_ignore,
+        LineSemantics.para_ignore_doc,
+        LineSemantics.para_doc,
+        LineSemantics.doc_ignore,
+    ]
+    for line_semantics in one_sentence:
+        documents = list(read_documents(test_document, line_sematics=line_semantics, lang=lang))
+        assert_document_counts(documents, 1, 1, 1, line_semantics)
+    for line_semantics in two_sentences:
+        documents = list(read_documents(test_document, line_sematics=line_semantics, lang=lang))
+        assert_document_counts(documents, 1, 1, 2, line_semantics)
+    assert len(LineSemantics) == len(two_sentences + one_sentence)
+
+
+def test_read_documents_bug_fix_1():
```

Comment:
bug_fix_1 er ekki mjög lýsandi nafn. Hvað var buggið? Hvernig var það lagað? Annaðhvort breyta nafninu til að lýsa testcase-inu eða ef þarf langa lýsingu þá í docstring: 
"""
Had a bug in jsonl parsing when the document ...
"""